lib/scss_lint/linter/private_naming_convention.rb

```diff
@@ -82,5 +131,27 @@ def after_visit_all
         end
       end
     end
+
+    def humanize_node_class(node)
+      case node
+      when Sass::Tree::FunctionNode
+        'function'
+      when Sass::Tree::MixinDefNode
+        'mixin'
+      when Sass::Tree::VariableNode
+        'variable'
+      end
+    end
+
+    def defining_class_for(node)
+      case node
+      when Sass::Script::Tree::Funcall
+        Sass::Tree::FunctionNode
+      when Sass::Tree::MixinNode
+        Sass::Tree::MixinDefNode
+      when Sass::Script::Tree::Variable
+        Sass::Tree::VariableNode
+      end
+    end
```

Comment:
You can combine these into a single hash, i.e. repurpose `DEFINING_CLASSES` into something like (forgive the naming):

```ruby
DEFINITIONS = {
  Sass::Tree::FunctionNode => { use_type: Sass::Script::Tree::Funcall, human_name: 'function' },
  ...,
}.freeze

DEFINITION_NAMES = DEFINITIONS.map { |k v| [k, v[:human_name] }.freeze
DEFINITION_TYPES = DEFINITION_TYPES.map { |k, v| [v[:use_type], k] }.freeze
```

Then these implementations become:

```ruby
def humanize_node_class(node)
  DEFINITION_NAMES[node]
end

def defining_class_for(node)
  DEFINITION_TYPES[node]
end
```

The advantage being you define all types once, rather than having to list these out in multiple places in verbose `case` statements. Not a big deal.